src/app/profile/payment-methods/recurring-gifts/recurring-gifts.component.js

```diff
@@ -0,0 +1,53 @@
+import angular from 'angular';
+import template from './recurring-gifts.tpl';
+import forEach from 'lodash/forEach';
+
+class recurringGiftsController{
+
+  /* @ngInject */
+  constructor(){}
+
+  $onInit(){
+    this.buildGifts();
+  }
+
+  buildGifts(){
+    var gifts = [];
+    forEach(this.gifts,(gift) => {
+      var rate = gift.rate,
+          day = gift['recurring-day-of-month'];
+      forEach(gift['donation-lines'], (donationLine) => {
+        donationLine['rate'] = rate;
+        donationLine['recurring-day-of-month'] = day;
+        gifts = gifts.concat(donationLine);
+      });
+    });
+    return gifts;
+  }
+
+  getNextGiftDate(gift){
```

Comment:
Could Adam's code https://github.com/CruGlobal/give-web/blob/master/src/common/services/api/cart.service.js#L153 work here? If you want to use it I guess it might be good to put it in a more general service that is not tied to the cart.

You can have angular format the date in the template: https://github.com/CruGlobal/give-web/blob/master/src/app/cart/cart.tpl.html#L35